tools/ci_build/github/linux/ort_minimal/readelf_utils.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+'''
+Utilities to help analyze the sections in a binary using readelf.
+'''
+
+import argparse
+import collections
+import os
+import re
+import sys
+import subprocess
+
+
+def get_section_sizes(binary_path, readelf_path, dump_to_file=None):
+    '''
+    Get the size of each section using readelf.
+    :param binary_path: Path to binary to analyze.
+    :param readelf_path: Path to readelf binary. Default is 'readelf'.
+    :param dump_to_file: File object to write section sizes and diagnostic info to. Defaults to None.
+    :return:
+    '''
+
+    cmd = [readelf_path, '--sections', '--wide', binary_path]
+    result = subprocess.run(cmd, stdout=subprocess.PIPE)
+    result.check_returncode()
+    output = result.stdout.decode('utf-8')
+
+    section_sizes = {}
+
+    # Parse output in this format:
+    #   [Nr] Name              Type            Address          Off    Size   ES Flg Lk Inf Al
+    for match in re.finditer(r'\[[\s\d]+\] (\..*)$', output, re.MULTILINE):
+        items = match.group(1).split()
+        name = items[0]
+        # convert size from hex to int
+        size = int(items[4], 16)
+        section_sizes[name] = size
+
+        if dump_to_file:
+            print('{}:{}'.format(name, size), file=dump_to_file)
+
+    return section_sizes
+
+
+def get_sections_total_size(binary_path, readelf_path='readelf'):
```

Comment:
>get_sections_total_size [](start = 4, length = 23)

Seems not used any where, still need this?
Or add an extra param dump_to_file=None, and call get_sections_total_size instead of get_section_sizes then sum in check_build_binary_size?

Reply:
I was using it and changed to get the dump output. I'll remove it. It's a bit cleaner for a user who wants to dump values to use the lower level function than to add a dump_to_file= at another level.